layers/core_checks/cc_buffer_address.h

```
@@ -125,7 +151,26 @@ bool BufferAddressValidation<N>::HasValidBuffer(
 }
 
 template <size_t N>
-bool BufferAddressValidation<N>::LogInvalidBuffers(const CoreChecks& checker, const LogObjectList& objlist,
+bool BufferAddressValidation<N>::HasInvalidBuffer(
+    vvl::span<const ValidationStateTracker::BUFFER_STATE_PTR> buffer_list) const noexcept {
+    for (const auto& buffer : buffer_list) {
+        assert(buffer);
```

I guess what is the point of having an `assert` if we have this `continue` case... should only be one

---

I want the `assert` to catch problems in debug builds, and a `continue` to make sure the program does not crash when dereferencing a null pointer in release builds

---

So I guess my question is are `null` pointers "suppose" to happen?

If this returns null in a release build, will it just cause a stranger error in a few function calls, making it hard to track down this is where the problem happened

---

I guess my concern is we just had a discussion about the fact we are always checking if `Get<>` is returning null, but it becomes hard quick to know if things "can" return null, or if it does, everything is screwed anyway

I guess, I will let you decide because I fully haven't grasped `LogErrorsIfInvalidBufferFound` and `HasInvalidBuffer`, but if things are null, we should either have some error or just assert... I don't like the idea of silently hiding it without a comment at least explaining why we can just ignore it

---

Then I am going to leave the `assert` and remove the `continue`